server/autocomplete_search.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+func httpAPIGetAutoCompleteFields(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodGet {
+		return http.StatusMethodNotAllowed,
+			errors.New("Request: " + r.Method + " is not allowed, must be GET")
+	}
+
+	mattermostUserID := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserID == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	jiraUser, err := ji.GetPlugin().userStore.LoadJIRAUser(ji, mattermostUserID)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	client, err := ji.GetClient(jiraUser)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	params := map[string]string{
+		"fieldName":  r.FormValue("fieldName"),
+		"fieldValue": r.FormValue("fieldValue"),
+	}
+
+	results, err := client.SearchAutoCompleteFields(params)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	if results == nil {
```

Comment:
@larkox I'm actually not sure if this block will ever run. If there are just no results that match the user's search, it returns an empty array instead of `nil`. This was taken from a contributor's stale PR, and I didn't change this block after adopting the code.